torch/csrc/jit/pickler.cpp

```diff
@@ -786,18 +808,46 @@ PickleOpCode Unpickler::readInstruction() {
           int64_t storage_offset = elements.at(idx++).toInt();
           std::vector<int64_t> size = tupleToIntList(elements.at(idx++));
           std::vector<int64_t> stride = tupleToIntList(elements.at(idx++));
-          double q_scale = 0.;
-          int64_t q_zero_point = 0;
+          at::Tensor result;
           if (quantized) {
-            q_scale = elements.at(idx++).toDouble();
-            q_zero_point = elements.at(idx++).toInt();
+            auto qparams_tuple = elements.at(idx++).toTuple();
+            const auto& qparams = qparams_tuple->elements();
+            auto qscheme = static_cast<at::QScheme>(qparams.at(0).toInt());
+            switch (qscheme) {
+              case at::kPerTensorAffine: {
+                double q_scale = qparams.at(1).toDouble();
+                int64_t q_zero_point = qparams.at(2).toInt();
+                result = at::_empty_affine_quantized(
+                    {0}, storage_tensor.options(), q_scale, q_zero_point);
+              } break;
+              case at::kPerChannelAffine: {
```

Comment:
We should also handle kPerChannelSymmetric and kPerTensorSymmetric here

Reply:
again, we have only two types of quantizers present today